### PR TITLE
Add Particular.CodeRules 0.4.0

### PR DESCRIPTION
**Pull request by kbaley**

### CODE REVIEW

File: src/ServiceInsight.Tests/AboutViewModelTests.cs
Comment (adamralph) [inline src/ServiceInsight.Tests/AboutViewModelTests.cs:37]: This won't fix this test. This test was never proving anything, since the call to `GetVersion` is fire-and-forget. The correct fix is:

```C#
        [Test]
        public async Task Should_fetch_service_control_version()
        {
            ((IActivate)sut).Activate();

            await serviceControl.Received(1).GetVersion();
        }
```

or

```C#
        [Test]
        public Task Should_fetch_service_control_version()
        {
            ((IActivate)sut).Activate();

            return serviceControl.Received(1).GetVersion();
        }
```

I'll push a commit which fixes these.

**Comment by adamralph** [comment]
@kbaley I pushed a few more commits. Can you please take a look?

If you agree with them, I think we should squash all these commits into one before merging. I don't think any of them have independent value. Arguably, the R# removal could be split from the addition of CodeRules, but I don't think it's important.

**Comment by kbaley** [comment]
@adamralph These look good to me. I can't merge without an approving review and I can't approve my own PR so go ahead and squash these and merge when you can.

**Review by adamralph — APPROVED** [review]

**Pull request merged**
